design/one-pager-prevent-unnecessary-reconciles.md

```diff
@@ -0,0 +1,53 @@
+# Prevent unnecessary reconciles
+* Owner: Christian Artin (@gravufo)
+* Reviewers: Crossplane Maintainers
+* Status: Speculative
+
+## Introduction
+
+Today, Crossplane providers will reconcile resources every time there is a change, regardless of the change, _and_ at pod startup _and_ when the `SyncInterval` is hit.
+While this does not seem too problematic at a glance, it becomes very problematic when there are thousands of objects managed by that provider and when each reconcile can make one or multiple calls to the external provider.
+Side-effects include slow performance, external rate-limiting and higher CPU usage.
+A common trigger to this issue is a simple update of the provider to a newer version which causes a new pod to start which will then reconcile all its managed resources, even if they were reconciled a minute before.
+
+## Definitions
+
+- External reconciliation: A reconciliation which does actual work (i.e.: calls the external provider to get the resource's actual status)
+
+## Goal
+
+Significantly reduce the number of external reconciles done on resources to:
+- Prevent unnecessary CPU usage
+- Make new pods work on actual changes faster instead of being stuck behind a queue of resources that do not need to be re-validated
+- Reduce load on external providers thus reducing chances of being rate limited
+
+Reduce confusion between `SyncInterval` and `PollInterval`. With the proposed changes, we will be able to clearly define both of those properties as follows:
+- `SyncInterval`: At which frequency resources should be reconciled externally to ensure the desired state is still applied.
+- `PollInterval`: At which frequency the provider should re-list all its watched resources from the Kubernetes API to ensure the cache is up-to-date and no event was missed. Irrelevant when real-time compositions are enabled.
+
```

Comment:
This sounds flipped relative to what they are today. Is that intentional?

Today sync interval sets the [controller-runtime cache `SyncPeriod` setting](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.4/pkg/cache#Options). It's the interval at which the underlying cache for a particular kind of resource resists all instances of that kind of resource.

Poll interval on the other hand is the interval at which a specific instance of a resource is checked for drift.

Reply:
You're probably right! Goes to show that this is definitely a source of confusion :)
I will fix that.